Include entry points as an optional feature (#297)
data-apis/array-api: spec/purpose_and_scope.md

harder for users to write code that will work with multiple array libraries.

### Discoverability

> 

Reviewer: This is a bit cryptic for a reader, like why does it exist? what goes wrong if an entrypoint is not implemented? The one word _may_ is not enough, I'll try to add a note with rationale. Especially because this is in the "Purpose and scope" section, that is important.

Reviewer 2: Thanks!

> To assist array-consuming libraries which need to create arrays originating from multiple conforming array implementations, conforming implementations may provide a {pypa}`entry point <specifications/entry-points/>` in order to make an array API namespace discoverable:

Reviewer 2: ```suggestion
To assist array-consuming libraries which need to create arrays originating from multiple conforming array implementations, conforming implementations may provide an {pypa}`entry point <specifications/entry-points/>` in order to make an array API namespace discoverable:
```

> To assist array-consuming libraries which need to create arrays originating from multiple conforming array implementations, conforming implementations may provide an {pypa}`entry point <specifications/entry-points/>` in order to make an array API namespace discoverable:

Reviewer 2: ```suggestion
To assist array-consuming libraries which need to create arrays originating from multiple conforming array implementations, conforming implementations may provide a {pypa}`entry point <specifications/entry-points/>` in order to make an array API namespace discoverable:
```

> To assist array-consuming libraries which need to create arrays originating from multiple conforming array implementations, conforming implementations may provide an {pypa}`entry point <specifications/entry-points/>` in order to make an array API namespace discoverable. For example,

Reviewer 2: Thanks for looking over!

harder for users to write code that will work with multiple array libraries.

### Discoverability

> Libraries that wish to [create arrays](../API_specification/creation_functions.md)
> using multiple array libraries would still need to explicitly import each
> library, as [`__array_namespace__()`](method-__array_namespace__) only makes an
> API namespace accessible when consuming arrays.
> 
> To address this issue, an {pypa}`entry point <specifications/entry-points/>` may
> be provided by a conforming implementation to make its API namespace
> discoverable:

Reviewer 2: ```suggestion
To assist array-consuming libraries which need to create arrays originating from multiple conforming array implementations, conforming implementations may provide an {pypa}`entry point <specifications/entry-points/>` in order to make an array API namespace discoverable:
```

```python
from importlib.metadata import entry_points

try:
    eps = entry_points()['array_api']
    ep = next(ep for ep in eps if ep.name == 'package_name')
except TypeError:
    # The dict interface for entry_points() is deprecated in py3.10,
    # supplanted by a new select interface.
    ep = entry_points(group='array_api', name='package_name')

xp = ep.load()
```

> The properties of an entry point should have:
> 
> - `array_api` as the **group**
> - The package name as the **name**
> - The import path to the API namespace as the **object reference**

Reviewer 2: ```suggestion
An entry point must have the following properties:

-   **group**: equal to `array_api`.
-   **name**: equal to the package name.
-   **object reference**: equal to the array API namespace import path.
```
